src/CSharp/CodeCracker/Refactoring/XmlDocumentationAnalyzer.cs

```diff
@@ -0,0 +1,65 @@
+﻿using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Diagnostics;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+
+namespace CodeCracker.CSharp.Refactoring
+{
+    [DiagnosticAnalyzer(LanguageNames.CSharp)]
+    public class XmlDocumentationAnalyzer : DiagnosticAnalyzer
+    {
+        internal const string Title = "You have missing/unexistent parameters in Xml Docs";
+        internal const string MessageFormat = "You have missing/unexistent parameters in Xml Docs";
+        internal const string Category = SupportedCategories.Refactoring;
+
+        internal static DiagnosticDescriptor XmlDocumentationRule = new DiagnosticDescriptor(
+            DiagnosticId.XmlDocumentation.ToDiagnosticId(),
+            Title,
+            MessageFormat,
+            SupportedCategories.Refactoring,
+            DiagnosticSeverity.Warning,
+            isEnabledByDefault: true,
+            helpLinkUri: HelpLink.ForDiagnostic(DiagnosticId.XmlDocumentation));
+
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(XmlDocumentationRule);
+
+        public override void Initialize(AnalysisContext context) => context.RegisterSyntaxNodeAction(Analyzer, SyntaxKind.SingleLineDocumentationCommentTrivia);
+
+        private static void Analyzer(SyntaxNodeAnalysisContext context)
+        {
+            var documentationNode = (DocumentationCommentTriviaSyntax)context.Node;
+            var method = documentationNode.AncestorsAndSelf().OfType<MethodDeclarationSyntax>().First();
+            var methodParameters = method.ParameterList.Parameters;
+            var xElementsWitAttrs = documentationNode.Content.OfType<XmlElementSyntax>()
+                                    .Where(xEle => xEle.StartTag.Name.LocalName.ValueText == "param")
+                                    .SelectMany(xEle => xEle.StartTag.Attributes, (xEle, attr) => (XmlNameAttributeSyntax)attr);
+
+            var methodParameterLookUp = methodParameters.ToLookup(parameter => parameter.Identifier.ValueText);
+            var docParameterLookUp = xElementsWitAttrs.ToLookup(attr => attr.Identifier.Identifier.ValueText);
+
+            var keys = methodParameterLookUp.Select(p => p.Key).Union(docParameterLookUp.Select(p => p.Key)).ToImmutableHashSet();
```

Comment:
It does not seem necessary to create 2 lookups if the only reason for their existence is to later create a union on their keys. Analyzers are all about performance, simply create a hash from the parameter identifiers and xelements without creating the intermediate lookup.